Main.elm

```diff
@@ -1,7 +1,393 @@
+    , kintoCollection : String
+    }
+
+
+type alias Bearer =
+    String
+
+
+type alias RecordId =
+    String
+
+
+type alias Record =
+    { id : RecordId
+    , last_modified : Int
+    , subject : Maybe String
+    , stateMachineArn : String
+    , activityArn : String
+    , status : Maybe String
+    }
+
+
+
+-- Msg
+
+
+type Msg
+    = NewEmail String
+    | Authenticate
+    | LoadRecords
+    | FetchRecordsResponse (Result Kinto.Error (List Record))
+    | Logout
+    | AcceptStep String
+    | RejectStep String
+    | AnswerResponse (Result Kinto.Error ())
+    | Tick Time
+
+
+
+-- Init
+
+
+init : Flags -> ( Model, Cmd Msg )
+init flags =
+    let
+        model =
+            { email = Maybe.withDefault "" flags.email
+            , bearer = flags.bearer
+            , records = Nothing
+            , error = Nothing
+            , kintoServer = flags.kintoServer
+            , kintoBucket = flags.kintoBucket
+            , kintoCollection = flags.kintoCollection
+            }
+    in
+        update LoadRecords model
+
+
+
+-- Subscriptions
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Sub.batch [ Time.every minute Tick ]
+
+
+
+-- Updates
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update message model =
+    case message of
+        NewEmail email ->
+            { model | email = email }
+                ! [ saveData { key = "email", value = Encode.string email } ]
+
+        Authenticate ->
+            model ! [ authenticate model.email ]
+
+        LoadRecords ->
+            case model.bearer of
+                Nothing ->
+                    model ! []
+
+                Just _ ->
+                    { model
+                        | records = Just []
+                        , error = Nothing
+                    }
+                        ! [ fetchRecordList model ]
+
+        FetchRecordsResponse (Ok recordList) ->
+            { model
+                | records = Just recordList
+                , error = Nothing
+            }
+                ! []
+
+        FetchRecordsResponse (Err error) ->
+            { model | error = Just <| toString error } ! []
+
+        Logout ->
+            { model
+                | error = Nothing
+                , records = Nothing
+                , bearer = Nothing
+            }
+                ! [ saveData { key = "bearer", value = Encode.null } ]
+
+        AcceptStep recordId ->
+            model ! [ answerStep model recordId "SUCCEED" ]
+
+        RejectStep recordId ->
+            model ! [ answerStep model recordId "FAIL" ]
+
+        AnswerResponse (Err error) ->
+            { model | error = Just <| toString error } ! []
+
+        AnswerResponse (Ok _) ->
+            model ! [ fetchRecordList model ]
+
+        Tick _ ->
+            model ! [ fetchRecordList model ]
+
+
+
+-- Kinto related
+
+
+answerUrl : String -> String -> String -> String -> Url
+answerUrl baseUrl bucketName collectionName recordId =
+    let
```

Comment:
I don't think you can extend it. We could get the record url and then we would have to add stepfunction at the end.